src/Security/XpubTokenGenerator.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace eLife\Journal\Security;
+
+use Firebase\JWT\JWT;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+final class XpubTokenGenerator
+{
+    const TOKEN_TTL = 60;
+
+    private $clientId;
+    private $clientSecret;
+
+    public function __construct(string $clientId, string $clientSecret)
+    {
+        $this->clientId = $clientId;
+        $this->clientSecret = $clientSecret;
+    }
+
+    public function generate(UserInterface $user) : string
+    {
+        return JWT::encode([
+            'iss' => $this->clientId,
+            'iat' => time(),
+            'exp' => time() + self::TOKEN_TTL,
+            'id' => $user->getUsername(),
```

Comment:
I accidentally set `iss` to `xpub` but if you want `journal--{instance}` it can be changed:
https://github.com/elifesciences/builder-private/pull/251/files
I assume that then Xpub doesn't use this field at all and only validates with the configured `PUBSWEET_SECRET`?

Reply:
Up to @diversemix/@tamlyn.

Reply:
This is config so ignoring for now.

Reply:
@thewilkybarkid  / @giorgiosironi The important thing is that its not set to `xpub` ... when the token is received `xpub` checks that it has not issued the token (i.e. its not `iss == xpub`) and then the exchange takes place.

Reply:
Opened https://github.com/elifesciences/builder-private/pull/252.